test/integ/pandas/test_arrow_pandas.py

```diff
@@ -1202,6 +1202,89 @@ def test_batch_to_pandas_arrow(conn_cnx, result_format):
                 assert arrow_table.to_pydict()["FOO"] == list(range(rowcount))
 
 
+@pytest.mark.internal
+@pytest.mark.parametrize("enable_structured_types", [True, False])
+def test_to_arrow_datatypes(enable_structured_types, conn_cnx):
+    with conn_cnx() as cnx:
+        with cnx.cursor() as cur:
+            cur.execute(SQL_ENABLE_ARROW)
+
+            expected_types = (
+                pyarrow.int64(),
+                pyarrow.float64(),
+                pyarrow.string(),
+                pyarrow.date64(),
+                pyarrow.timestamp("ns"),
+                pyarrow.string(),
+                pyarrow.timestamp("ns"),
+                pyarrow.timestamp("ns"),
+                pyarrow.timestamp("ns"),
+                pyarrow.binary(),
+                pyarrow.time64("ns"),
+                pyarrow.bool_(),
+                pyarrow.string(),
+                pyarrow.string(),
+                pyarrow.list_(pyarrow.float64(), 5),
+            )
+
+            if enable_structured_types:
+                for param in {
+                    "ENABLE_STRUCTURED_TYPES_IN_CLIENT_RESPONSE",
+                    "IGNORE_CLIENT_VESRION_IN_STRUCTURED_TYPES_RESPONSE",
+                    "FORCE_ENABLE_STRUCTURED_TYPES_NATIVE_ARROW_FORMAT",
+                }:
```

Comment:
can we unset them afterwards or use `_statement_params` to populate these values so that this change is contained in here